### PR TITLE
feat(files): rework store nsfw, scan consent, update diff instances

### PR DESCRIPTION
**Pull request by josephmcg**
🔴 BREAKING CHANGE - you need a new account to test. I refactored the UserInfoManager to use camelCase for the json schema to be easier to work with in the rest of our app. it won't be able to pull the data properly if you already have a record

**What this PR does** 📖
- update textile bucket index with vuex action. In that same action, update file system version number in threaddb via UserInfoManager 
- refactor UserInfoManager to simplify update logic
- improve typescript doc and fix intellisense on a few files
- move nsfwBlock and consentScan to a different part of the store, keeping track of the entire threaddb record as one entity (textile.threadData)
- setup listener to update textile.threadData if the value changes on any instance of this account
- don't allow file uploads (file system page and chat) or settings updates until textile is done with init. I do this because the upload process needs to be aware of the scan consent setting (stored in textile)

**Which issue(s) this PR fixes** 🔨
AP-1522, AP-1523

**Special notes for reviewers** 🗒️
- new major functionality is updating settings across instances of the same account. If you login to the same account in two different tabs/windows, this will update your scan consent and nsfw blocker settings across both instances.

**Additional comments** 🎤

### CODE REVIEW

**Review by phillsatellite — APPROVED** [review]

**Comment by josephmcg** [comment]
@phillsatellite , I improved the ux when you toggle those two settings. now it gets disabled while it's updating the remote textile value and provides a status message at the bottom

**Comment by josephmcg** [comment]
@stavares843 thanks, I broke it with that ui refactor. should be better now

**Comment by iltumio** [comment]
it looks like we are still using the old instance of textile hub for buckets:

BucketManager.ts line 51. Can you have a look at it?
```
    const result = await this.buckets.getOrCreate(
      `hub.textile.io/ipfs/${this.identity}/${this.prefix}`,
    )
```

**Comment by josephmcg** [comment]
> it looks like we are still using the old instance of textile hub for buckets:
> 
> 
> 
> BucketManager.ts line 51. Can you have a look at it?
> 
> ```typescript
> 
>     const result = await this.buckets.getOrCreate(
> 
>       `hub.textile.io/ipfs/${this.identity}/${this.prefix}`,
> 
>     )
> 
> ```

I can see why it appears that way, but that's just the cosmetic bucket name. Like how the personal files encrypted bucket is named 'personal-files'. I ultimately plan on switching chat to use the newer Bucket class

**Comment by iltumio** [comment]
Perfect. Thanks for the explanation. We should chose a better name that is agnostic to the platform we are using

File: store/textile/actions.ts [hunk 1]
```diff
@@ -1030,4 +1029,79 @@ export default {
...
+  /**
+   * @description listen for user data thread changes and update store accordingly
+   */
+  async listenToThread({ commit, rootState }: ActionsArguments<TextileState>) {
+    const $UserInfoManager: UserInfoManager =
+      Vue.prototype.$TextileManager?.userInfoManager
+    const $FileSystem: FilSystem = Vue.prototype.$FileSystem
+    const callback = (update?: Update<UserThreadData>) => {
+      if (!update || !update.instance) return
+      console.log('update', update.instance)
```
Comment (iltumio) [inline store/textile/actions.ts:1092]: remove console log

File: store/textile/actions.ts [hunk 2]
```diff
@@ -1030,4 +1029,79 @@ export default {
...
+      if (
+        update.instance.filesVersion &&
+        rootState.textile.userThread.filesVersion !== $FileSystem.version
+      ) {
+        console.log('todo - update file system based on bucket index')
```
Comment (iltumio) [inline store/textile/actions.ts:1097]: remove console log or use custom logger

File: store/ui/getters.test.ts [hunk 3]
```diff
@@ -10,9 +10,9 @@ describe('init', () => {
   })
 
   it('should return the isFilesIndexLoading property of the initial state', () => {
-    const result: any = inst.isFilesIndexLoading(InitialUIState())
-    expect(result).toBeFalsy()
-    expect(result).toMatchSnapshot()
+    // const result: any = inst.isFilesIndexLoading(InitialUIState())
```
Comment (iltumio) [inline store/ui/getters.test.ts:13]: should we keep this comment as reference for the QA team or remove them?

**Review by iltumio — CHANGES_REQUESTED** [review]
Looks like this PR is still waiting for Textile initialization

File: store/ui/getters.test.ts (hunk 3 above)
Comment (stavares843) [reply store/ui/getters.test.ts:13]: was this failing? @josephmcg i can ask andre to update this

**Comment by josephmcg** [comment]
yep, it's been sitting here for a week or two so I'm not surprised it's outdated. I'll rebase and clean up logs

image

File: store/ui/getters.test.ts (hunk 3 above)
Comment (josephmcg) [reply store/ui/getters.test.ts:13]: yeah i changed the getter to disable file system ui while textile is initializing. thanks for catching

File: libraries/Textile/UserManager.ts [hunk 4]
```diff
@@ -53,67 +53,62 @@ export class UserInfoManager {
...
+    if (!record) {
+      await this.textile.client.create(this.threadID, CollectionName, [
+        {
+          userAddress: this.textile.wallet.address,
+          consentToScan,
+          blockNsfw,
+          filesVersion,
+        },
+      ])
+      return await this._findRecord()
```
Comment (molimauro) [inline libraries/Textile/UserManager.ts:94]: return await redundant, simply return :)

File: store/ui/getters.test.ts (hunk 3 above)
Comment (josephmcg) [reply store/ui/getters.test.ts:13]: @stavares843 I would appreciate Andre's help in fixing this, thanks

**Review by molimauro — APPROVED** [review]

**Review by phillsatellite — APPROVED** [review]

**Review by stavares843 — CHANGES_REQUESTED** [review]
when sending a file on the chat example a zip returns always this error, doesn't happen on the dev branch

Captura de ecrã 2022-05-28, às 01 40 04

**Review by stavares843 — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/29093946/170803012-972d1f6b-7f50-4546-ba14-61d87f4efa87.mov

if you already gave the permissions to scan files, and the chat is loading and you try to upload a file will say to the user to consent the permission but the permission was already there

**Review by stavares843 — CHANGES_REQUESTED** [review]
Captura de ecrã 2022-05-28, às 01 01 52

minor but noticed the toggle appears all grey out when updating, considering is a toggle seems a layout bug

**Comment by stavares843** [comment]
to not block any further work I will get this merged and add tickets for the stuff above

**Comment by stavares843** [comment]
[AP-1653](https://satellite-im.atlassian.net/browse/AP-1653)
[AP-1654](https://satellite-im.atlassian.net/browse/AP-1654)
[AP-1655](https://satellite-im.atlassian.net/browse/AP-1655)

**Review by stavares843 — APPROVED** [review]

**Pull request merged**